bgpd/bgp_vty.c

```diff
@@ -3600,6 +3600,53 @@ DEFUN (bgp_default_shutdown,
 	return CMD_SUCCESS;
 }
 
+DEFPY (bgp_shutdown_msg,
+       bgp_shutdown_msg_cmd,
+       "bgp shutdown message MSG...",
+       BGP_STR
+       "Enable administrative shutdown of the BGP instance\n"
+       "Add a shutdown message (RFC 8203)\n"
+       "Shutdown message\n")
+{
+	char* msgstr = NULL;
+
+	VTY_DECLVAR_CONTEXT(bgp, bgp);
+
+	if (argc > 3)
+		msgstr = argv_concat(argv, argc, 3);
+
+	bgp_shutdown_enable(bgp, msgstr);
```

Comment:
can't we just iterate over the neighbors and mark them as shutdown `peer_flag_set(peer, PEER_FLAG_SHUTDOWN);` instead of introducing a global BGP flag?

Reply:
We could, but then we would have to introduce a way to memorize which peers were shut down before, as to be able to not lift their individual shutdown when we disable the global one.

Reply:
Makes sense.

Reply:
Also, this feature will most likely be used to do massive changes in the configuration without a restart of the daemon, in which case almost all peers would be affected anyway. If you think it would make sense, i could add a "ignore global shutdown" flag, which could be set by a separate command to exclude peers from the shutdown.

Reply:
but keep in mind that uint32_t only allows a limited amount of 1 bit flags to be stored.